packages/client/src/views/OfficeHome/readyForReview/ReadyForReview.tsx

```diff
@@ -82,87 +82,81 @@ interface IBaseReviewTabProps {
   error?: boolean
 }
 
-interface IReviewTabState {
-  width: number
-  sortedCol: COLUMNS
-  sortOrder: SORT_ORDER
-}
-
 type IReviewTabProps = IntlShapeProps & IBaseReviewTabProps
 
-class ReadyForReviewComponent extends React.Component<
-  IReviewTabProps,
-  IReviewTabState
-> {
-  pageSize = 10
-  constructor(props: IReviewTabProps) {
-    super(props)
-    this.state = {
-      width: window.innerWidth,
-      sortedCol: COLUMNS.SENT_FOR_REVIEW,
-      sortOrder: SORT_ORDER.DESCENDING
-    }
-  }
-
-  componentDidMount() {
-    window.addEventListener('resize', this.recordWindowWidth)
-  }
+const ReadyForReviewComponent = ({
+  theme,
+  scope,
+  goToPage,
+  goToDeclarationRecordAudit,
+  outboxDeclarations,
+  queryData,
+  paginationId,
+  onPageChange,
+  loading,
+  pageSize = 10,
+  error,
+  intl
+}: IReviewTabProps) => {
+  const [width, setWidth] = useState(window.innerWidth)
+  const [sortedCol, setSortedCol] = useState(COLUMNS.SENT_FOR_REVIEW)
+  const [sortOrder, setSortOrder] = useState(SORT_ORDER.DESCENDING)
 
-  componentWillUnmount() {
-    window.removeEventListener('resize', this.recordWindowWidth)
+  const recordWindowWidth = () => {
+    setWidth(window.innerWidth)
   }
 
-  recordWindowWidth = () => {
-    this.setState({ width: window.innerWidth })
-  }
+  useEffect(() => {
+    window.addEventListener('resize', recordWindowWidth)
+    return () => {
+      window.removeEventListener('resize', recordWindowWidth)
+    }
+  }, [])
```

Comment:
@Eezi Can we extract this out to a custom hook please? I'm pretty sure there are other components which will be using this

Reply:
Good idea! I can do that as well.